Add Visual Editor Docs (#248)
directus/docs: content/guides/2.content/8.visual-editor/0.index.md

Reviewer: In general, I am not sure about the term "frontend package". Is "website package" better? Or "website library"? Keep in mind, that this is not just an NPM package: PHP sites need to download the precompiled `/dist/visual-editing.js` file from GitHub (or NPM).

Reviewer 2: I would argue that the term "frontend package" being ubiquitous to this degree makes it a suitable term

Author: Yeah I think it works fine, "website" is too broad imo

Reviewer: The official description of the library is: “Visual editing library to enable in-place editing of your website’s frontend from within the Visual Editor in Directus”

So maybe we call it “Frontend Library”?

Reviewer: I think we should add an image or a video here. Maybe we can take this existing preview video until we have a better one? Let me know and I send you the HQ version of it.

https://github.com/user-attachments/assets/d3177ebb-79a9-495a-a4d5-173493697b71

Reviewer 2: I agree! This would be great

Author: Yeah great idea

---
title: Visual Editor

> description: The Directus Visual Editor module enables live editing of content directly in place.

Reviewer: Alternative:

The Directus Visual Editor module allows you to visually edit the content of your website directly in place.

Reviewer 2: ```suggestion
description: The Directus Visual Editor module allows you to visually edit the content of your website directly in-place.
```

I agree! Love this

---

> Making use of the Visual Editor requires some effort on the part of frontend developers through two separate but related aspects. These are the [Frontend Package](/guides/content/visual-editor/frontend-package), and the [Studio Module](/guides/content/visual-editor/studio-module) built into Directus.

Reviewer: website developers instead of frontend developers?

Reviewer 2: Again I think frontend is the more ubiquitous term here

Reviewer 2: How about web developers?

Author: Yeah I'm happy with "web developers"

> The package sets up the connection between your frontend website and your Directus instance. The studio module is then used to render your website within the Directus studio, make changes in place, and then render those changes as they are made.

Reviewer: ```suggestion
The package sets up the connection between your website and your Directus instance. The studio module is then used to render your website within the Directus studio, make changes in place, and then render those changes as they are made.
```

> ## Differences with Live Preview

Reviewer: Visual Editor does NOT behave similar to Live Preview – they both render your website in an iFrame.

@hola-soy-milk I am not sure about presenting this as a table. It takes up a lot of space on this page and it seems like we are afraid that people might get confused with Live Preview and Visual Editor.

Reviewer 2: How about a [collapsible](https://github.com/directus/docs/blob/065aafe675d7368d0e6223fdd4f8f699ffab5e48/content/releases/3.breaking-changes/1.version-10.md?plain=1#L45)? I do like the table

Author: I am open to whatever

---
title: Visual Editor
description: The Directus Visual Editor module allows you to visually edit the content of your website directly in-place.
---

Making use of the Visual Editor requires some effort on the part of frontend developers through two separate but related aspects. These are the [Frontend Package](/guides/content/visual-editor/frontend-package), and the [Studio Module](/guides/content/visual-editor/studio-module) built into Directus.

The package sets up the connection between your website and your Directus instance. The studio module is then used to render your website within the Directus studio, make changes in place, and then render those changes as they are made.

> :::collapsible{name="Differences with Live Preview" class="mt-2"}

Reviewer: I think we should only wrap the table into the collapsible.

> The Visual Editor behaves similarly to [Live Preview](/guides/content/live-preview) in that it renders content in an iframe, but with a few important differences.

Reviewer: The purple color highlights the term "Live Preview" and makes it over present here IMO. Can we remove the link?

```suggestion
The Visual Editor behaves similarly to Live Preview in that it renders content in an iFrame, but with a few important differences.
```

|                    | Live Preview | Visual Editor |
|--------------------|--------------|---------------|

> | **Entry Point**    | Item Form | Website rendered in Studio Module |

Reviewer: Please make sure that "Entry Point" does not break, by adding a non-breaking space, like `Entry&nbsp;Point`

```suggestion
| **Entry&nbsp;Point** | Item Form | Website rendered in Studio Module |
```
